packages/headless/src/app/recommendation-engine.ts/recommendation-engine.ts

```diff
@@ -0,0 +1,104 @@
+import {StateFromReducersMapObject} from '@reduxjs/toolkit';
+import {buildEngine, EngineOptions, ExternalEngineOptions} from '../engine';
+import {Engine} from '../headless-engine';
+import {RecommendationAppState} from '../../state/recommendation-app-state';
+import {debug, pipeline, searchHub} from '../reducers';
+import {
+  RecommendationEngineConfiguration,
+  recommendationEngineConfigurationSchema,
+} from './recommendation-engine-configuration';
+import {buildLogger} from '../logger';
+import {Logger} from 'pino';
+import {buildThunkExtraArguments} from '../thunk-extra-arguments';
+import {
+  NoopPostprocessFacetSearchResponseMiddleware,
+  NoopPostprocessQuerySuggestResponseMiddleware,
+  NoopPostprocessSearchResponseMiddleware,
+} from '../../api/search/search-api-client-middleware';
+import {NoopPreprocessRequestMiddleware} from '../../api/platform-client';
+import {NoopPreprocessRequest} from '../../api/preprocess-request';
+import {SearchAPIClient} from '../../api/search/search-api-client';
+
+const recommendationEngineReducers = {debug, pipeline, searchHub};
+type RecommendationEngineReducers = typeof recommendationEngineReducers;
+type RecommendationEngineState = StateFromReducersMapObject<
+  RecommendationEngineReducers
+> &
+  Partial<RecommendationAppState>;
+
+export interface RecommendationEngine
+  extends Engine<RecommendationEngineState> {}
+
+export interface RecommendationEngineOptions
+  extends ExternalEngineOptions<RecommendationEngineState> {
+  /**
+   * The recommendation engine configuration options.
+   */
+  configuration: RecommendationEngineConfiguration;
+}
+
+export function buildRecommendationEngine(
```

Comment:
Should be documented, no ?
+ Code samples should be adapted. This can be done in another PR.

Reply:
Yep, I added some annotations and will look at samples + extracting into the JSON in a future PR.